internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -739,6 +739,35 @@ func NewCoreContainer(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTa
 		}
 	}
 
+	disableBuiltInDiscovery := cr.Spec.TargetDiscoveryOptions != nil && cr.Spec.TargetDiscoveryOptions.BuiltInDiscoveryDisabled
+	if disableBuiltInDiscovery {
+		envs = append(envs, corev1.EnvVar{
+			Name:  "CRYOSTAT_DISABLE_BUILTIN_DISCOVERY",
+			Value: fmt.Sprintf("%v", disableBuiltInDiscovery),
+		})
+	}
+
+	useEmptyDir := cr.Spec.StorageOptions != nil && cr.Spec.StorageOptions.EmptyDir != nil && cr.Spec.StorageOptions.EmptyDir.Enabled
+	if !useEmptyDir {
+		envs = append(envs, corev1.EnvVar{
+			Name:  "JDBC_URL",
+			Value: "jdbc:h2:file:/opt/cryostat.d/conf.d/h2;INIT=create domain if not exists jsonb as other",
```

Comment:
Right! I will fix this :D

Reply:
Should we also need to set `JDBC` name and password?

Reply:
For `h2:mem` the user/pass are irrelevant. This database is in-memory only so when the Cryostat JVM shuts down the database also disappears entirely.

For `h2:file`, if the username or password changes then Cryostat can't open the database file for access. If the database file is expected to be garbage collected and deleted along with Cryostat being undeployed then in effect it is also irrelevant. If a database file is expected to be available to reuse between different Cryostat instances (delete and recreate CR) then both of those Cryostat instances need to use the same username and password. In this case the JMX credentials keyring passphrase would also need to be the same between these two Cryostat instances.

When we add Postgres support later it'll have the same requirements as `h2:file`, plus the user/pass need to be configurable by the user along with the JDBC URL pointing at that database so that the user can set up their own DB instance and configure it and give Cryostat access.

Reply:
Right, I think it would make sense to mount these env vars too so later with Postgres, we can just override them with specs in Cryostat CR.

Reply:
Should I use the values in test setup? Seems like its `sa` for username and empty for password.

Reply:
To clarify:

> if the username or password changes then Cryostat can't open the database file for access

When Cryostat creates the `h2:file` database it uses the JDBC user/pass it has at that time. If a later Cryostat instance comes along after a restart etc. then it must have he same JDBC user/pass settings in order to access that same database file.

The JDBC user/pass do not need to be the ones currently specified as defaults in StorageModule or as overrides in smoketest.sh, those can be considered as totally arbitrary.

Reply:
For now when they are simply hardcoded, maybe just the CR name as the username or something like that. The password can probably just be empty. There is nothing sensitive in the database file other than the JMX credentials, which are encrypted within the file anyway, so I don't think this is too critical at the moment.

Reply:
Okayy that makes sense! I could use the same value for both.